In [20]:
# from IPython.display import HTML

# HTML('''<script>
# code_show=true; 
# function code_toggle() {
#  if (code_show){
#  $('div.input').hide();
#  } else {
#  $('div.input').show();
#  }
#  code_show = !code_show
# } 
# $( document ).ready(code_toggle);
# </script>
# The raw code for this IPython notebook is by default hidden for easier reading.
# To toggle on/off the raw code, click <a href="javascript:code_toggle()">here</a>.''')

In [21]:
import pandas as pd
import numpy as np
import seaborn as sns
from matplotlib import pyplot as plt
%matplotlib inline

In [22]:
#We will exclude the last row of the dataset once it represents no information for the analysis
steam = pd.read_csv(r'./data/steam-200k.csv', usecols=[0,1,2,3])

# Dataset description and prep

This dataset is a list of user behaviors, with columns: user-id, game-title, behavior-name, value. The behaviors included are 'purchase' and 'play'. The value indicates the degree to which the behavior was performed - in the case of 'purchase' the value is always 1, and in the case of 'play' the value represents the number of hours the user has played the game.

First, let's check a few lines of the dataset

In [23]:
steam.head()

151603712  The Elder Scrolls V Skyrim  purchase    1.0
0  151603712  The Elder Scrolls V Skyrim      play  273.0
1  151603712                   Fallout 4  purchase    1.0
2  151603712                   Fallout 4      play   87.0
3  151603712                       Spore  purchase    1.0
4  151603712                       Spore      play   14.9

We see that our csv file has **no header**, so it's considering the first observation as it. Let's fix that by importing it again and adding our columns names!

In [24]:
steam = pd.read_csv(r'data/steam-200k.csv',
                 usecols=[0,1,2,3],
                 names = ['user_id', 'game', 'behavior', 'status'])

In [25]:
steam.head()

user_id                        game  behavior  status
0  151603712  The Elder Scrolls V Skyrim  purchase     1.0
1  151603712  The Elder Scrolls V Skyrim      play   273.0
2  151603712                   Fallout 4  purchase     1.0
3  151603712                   Fallout 4      play    87.0
4  151603712                       Spore  purchase     1.0

Much better!

Now, we see we have duplicate rows of the same game for the same user_id. This happens for the way the dataset is built. I would recommend considering removing the rows for *behavior* = 'purchase', once it just brings the same information of the column *play* (that is, the user has aquired the game in order to play it. Or could he have bought it and not played and the dataset would just show purchase and not the play line?

Let's check?

In [26]:
n_games_purchased = steam[steam['behavior'] == 'purchase']['behavior'].value_counts().sum()
n_games_played = steam[steam['behavior'] == 'play']['behavior'].value_counts().sum()
ratio_played_purchased = n_games_played/n_games_purchased

In [27]:
print(f'Number of games purchased: {n_games_purchased}')
print(f'Number of games played: {n_games_played}')
print(f'Ratio Played/Purchased: {round(ratio_played_purchased*100,2)}%')

Number of games purchased: 129511
Number of games played: 70489
Ratio Played/Purchased: 54.43%


Well, from here we can see that the ratio of games purchased and played is *54.43%*, this means almost half the games bought are not played.

In [28]:
ign = pd.read_csv(r'./data/ign.csv')

In [29]:
ign.head()

Unnamed: 0 score_phrase                                              title  \
0           0      Amazing                            LittleBigPlanet PS Vita   
1           1      Amazing  LittleBigPlanet PS Vita -- Marvel Super Hero E...   
2           2        Great                               Splice: Tree of Life   
3           3        Great                                             NHL 13   
4           4        Great                                             NHL 13   

                                                 url          platform  score  \
0             /games/littlebigplanet-vita/vita-98907  PlayStation Vita    9.0   
1  /games/littlebigplanet-ps-vita-marvel-super-he...  PlayStation Vita    9.0   
2                          /games/splice/ipad-141070              iPad    8.5   
3                      /games/nhl-13/xbox-360-128182          Xbox 360    8.5   
4                           /games/nhl-13/ps3-128181     PlayStation 3    8.5   

        genre editors_choice  release_year  release_month  release_day  
0  Platformer              Y          2012              9           12  
1  Platformer              Y          2012              9           12  
2      Puzzle              N          2012              9           12  
3      Sports              N          2012              9           11  
4      Sports              N          2012              9           11

In [30]:
#Creating PC only dataframe for the ign dataset
ign_pc = ign[ign['platform'] == 'PC'].copy()

In [31]:
ign_pc.head()

Unnamed: 0 score_phrase                            title  \
7            7      Amazing                     Guild Wars 2   
9            9         Good        Total War Battles: Shogun   
14          14      Amazing                Mark of the Ninja   
16          16         Okay  Home: A Unique Horror Adventure   
23          23         Good         Mass Effect 3: Leviathan   

                                                url platform  score  \
7                     /games/guild-wars-2/pc-896298       PC    9.0   
9         /games/total-war-battles-shogun/pc-142564       PC    7.0   
14        /games/mark-of-the-ninja-135615/pc-143761       PC    9.0   
16  /games/home-a-unique-horror-adventure/pc-137135       PC    6.5   
23         /games/mass-effect-3-leviathan/pc-138919       PC    7.5   

                genre editors_choice  release_year  release_month  release_day  
7                 RPG              Y          2012              9           11  
9            Strategy              N          2012              9           11  
14  Action, Adventure              Y          2012              9            7  
16          Adventure              N          2012              9            6  
23                RPG              N          2012              8           31

In [32]:
steam.isna().sum()

user_id     0
game        0
behavior    0
status      0
dtype: int64

In [33]:
ign.isna().sum()

Unnamed: 0         0
score_phrase       0
title              0
url                0
platform           0
score              0
genre             36
editors_choice     0
release_year       0
release_month      0
release_day        0
dtype: int64

In [34]:
filtro = ign['genre'].isna()
ign[filtro]

Unnamed: 0 score_phrase  \
12             12         Good   
113           113         Good   
160           160         Good   
176           176         Okay   
9375         9375        Great   
9488         9488         Okay   
9767         9767         Good   
9774         9774      Amazing   
10494       10494        Awful   
11367       11367      Painful   
11513       11513         Good   
11699       11699     Mediocre   
11719       11719        Great   
12316       12316          Bad   
12620       12620         Good   
12865       12865         Okay   
13527       13527        Great   
13754       13754         Good   
14070       14070         Good   
14099       14099        Great   
14259       14259        Great   
14729       14729         Good   
14769       14769         Okay   
14911       14911         Okay   
15279       15279         Good   
15552       15552        Great   
15578       15578         Good   
15579       15579         Good   
15580       15580         Good   
15589       15589        Great   
15608       15608         Good   
16136       16136     Mediocre   
16179       16179         Good   
16953       16953        Great   
17686       17686        Awful   
17705       17705         Good   

                                                   title  \
12                                            Wild Blood   
113                                          Retro/Grade   
160                                             10000000   
176                                          Colour Bind   
9375                                    Duke Nukem Arena   
9488                                             Rengoku   
9767                                      Super Sketcher   
9774                                      Critter Crunch   
10494       Clue / Mouse Trap / Perfection / Aggravation   
11367                                       Jeep Thrills   
11513                                   21 Pro Blackjack   
11699                                       Golden Skull   
11719          Nancy Drew: The Haunting of Castle Malloy   
12316                                     PopStar Guitar   
12620                                         Heavy Mach   
12865                                      Tower Toppler   
13527                                Eyegore's Eye Blast   
13754                         Aha! I Got It! Escape Game   
14070                                         ShadowPlay   
14099                               The Horrible Vikings   
14259                                           The Hero   
14729                                           Primrose   
14769                                          AquaSpace   
14911                              Meow Meow Happy Fight   
15279                Stenches: A Zombie Tale of Trenches   
15552                                       Land-a Panda   
15578                     Naruto Shippuden: Kizuna Drive   
15579    Nintendogs + Cats: French Bulldog & New Friends   
15580        Nintendogs + Cats: Toy Poodle & New Friends   
15589                                          Liqua Pop   
15608  Nintendogs + Cats: Golden Retriever & New Friends   
16136                         Crystal Caverns of Amon-Ra   
16179                         Aya and the Cubes of Light   
16953                                    Impossible Road   
17686                                 War of the Vikings   
17705                                         Boom Beach   

                                                     url  \
12                       /games/wild-blood/iphone-139363   
113                   /games/retrograde-138590/ps3-21766   
160                        /games/10000000/iphone-139135   
176                         /games/colour-bind/pc-143757   
9375                 /games/duke-nukem-arena/cell-893821   
9488                          /games/rengoku/cell-924924   
9767                   /games/super-sketcher/cell-874054   
9774                   /games/critter-crunch/ce

In [ ]:
#mudança no master